src/Nethermind/Nethermind.Db/DbSpanMemoryManager.cs

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Nethermind.Db;
+
+namespace Nethermind.Core.Buffers;
+
+public class DbSpanMemoryManager : MemoryManager<byte>
+{
+    private readonly IDbWithSpan _db;
+    private readonly unsafe void* _ptr;
+    private readonly int _length;
+
+    public DbSpanMemoryManager(IDbWithSpan db, Span<byte> span)
+    {
+        unsafe
+        {
+            _db = db;
+            _ptr = Unsafe.AsPointer(ref span.GetPinnableReference());
```

Comment:
Should use `MemoryMarshal.GetReference` as `GetPinnableReference` will use a `null` ptr for `.Length == 0` whereas we want to keep whatever the db returns